client/my-sites/patterns/hooks/use-record-patterns-event.tsx

```diff
@@ -0,0 +1,20 @@
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { usePatternsContext } from 'calypso/my-sites/patterns/context';
+import { getTracksPatternType } from 'calypso/my-sites/patterns/lib/get-tracks-pattern-type';
+
+export const useRecordPatternsEvent = () => {
+	const { category, patternTypeFilter } = usePatternsContext();
+
+	return {
+		recordPatternsEvent: (
+			tracksEventName: string,
+			params: Record< string, string | number | undefined > = {}
+		) => {
+			recordTracksEvent( tracksEventName, {
+				category: category || undefined,
+				type: getTracksPatternType( patternTypeFilter ),
```

Comment:
```suggestion
				view: isGridView ? 'view' : 'list',
```
I just came to realize that, we only use this hook in pages other than the main page `/patterns`. So it's fine to default `view` property to `list` (i.e. default to `list` in search result).

Reply:
I it unintentionally deleted `type: getTracksPatternType( patternTypeFilter ),`? Shoudl we keep both `type` and `view`?

Reply:
No, I didn't mean to delete `type`. Sorry!